src/io.rs

```diff
@@ -1255,17 +1255,146 @@ pub trait WriteBytesExt: io::Write {
         T::write_f64(&mut buf, n);
         self.write_all(&buf)
     }
+
+    /// Writes a slice of i8 to the underlying writer
+    ///
+    /// # Errors
+	///
+	/// This method returns the same errors as [`Write::write_all`]
+	///
+    /// [`Write::write_all`]: https://doc.rust-lang.org/std/io/trait.Write.html#method.write_all
+    fn write_i8_from<T: ByteOrder>(&mut self, src: &[i8]) -> Result<()> {
+    {
+        let buf = unsafe { slice_to_u8(src) };
+        try!(self.write_all(buf));
+    }
+    Ok(())
+    }
+
+    /// Writes a slice of u8 to the underlying writer
+    fn write_u8_from<T: ByteOrder>(&mut self, src: &[u8]) -> Result<()> {
+    {
+        try!(self.write_all(src));
+    }
+    Ok(())
+    }
+
+    /// Writes a slice of i16 to the underlying writer
+    fn write_i16_from<T: ByteOrder>(&mut self, src: &[i16]) -> Result<()> {
+    {
+        let mut src_copy: Vec<i16> = vec![0; src.len()];
+        src_copy.copy_from_slice(src);
+        T::from_slice_i16(&mut src_copy);
+        let buf = unsafe { slice_to_u8(&src_copy) };
+        try!(self.write_all(buf));
+    }
+    Ok(())
+    }
```

Comment:
If this is the best possible implementation of this method, then I don't think it should be in `byteorder`. Creating a copy of the input data is completely unexpected.

Reply:
@BurntSushi I know another method I could try but then the slice parameter of these functions would be mutable - I dont know if that is acceptable.

Reply:
Yeah I thought of that too, but it kind takes the wind out of these functions such that I'm not sure it's worth having them. :-(